### PR TITLE
Reorganize docs

### PR DESCRIPTION
**Pull request by tylerbutler**
- (Re)created concepts section and moved some deep articles to the new section.
- Hid the depths section from nav.
- Moved React article to recipes section.

### CODE REVIEW

File: docs/.gitignore [hunk 1]
```diff
@@ -11,5 +11,7 @@ themes/thxvscode/assets/js/node_modules/
 
 static/playground/
 
-# Images in these folder are autogenerated at docs build time
-content/docs/*/images/**/*.png
+# Images at these paths are autogenerated at docs build time
+content/community/doc-system/*.png
+static/images/architecture.png
```
Comment (heliocliu) [inline docs/.gitignore:16]: seems like generated images here should go in their own folder so they don't need to all be explicitly added here (also having generated images under the static folder seems kind of off?)

File: docs/config.yml [hunk 2]
```diff
@@ -75,8 +75,8 @@ menu:
   - identifier: "DDS"
     name: "Data Structures"
     weight: -300
-  - identifier: "deep"
```
Comment (heliocliu) [inline docs/config.yml:78]: what's happening to the articles that remain under deep?  it doesn't look like there's a way to access all of them anymore with the new organization

File: docs/.gitignore (hunk 1 above)
Comment (tylerbutler) [reply docs/.gitignore:16]: `static` is [what Hugo calls the folder](https://gohugo.io/content-management/static-files/) of stuff that gets copied as-is to the output directory. I could move the source ditaa files to a `generated` folder, but then you have to refer to the output images with that in the URL.

File: docs/config.yml (hunk 2 above)
Comment (tylerbutler) [reply docs/config.yml:78]: Correct, they're all hidden for now. Consensus seems to be that they're confusing.

**Review by heliocliu — APPROVED** [review]

**Pull request merged**